src/epplibwrapper/utility/pool.py

```diff
@@ -0,0 +1,133 @@
+import logging
+from typing import List
+import gevent
+from geventconnpool import ConnectionPool
+from epplibwrapper.socket import Socket
+from epplibwrapper.utility.pool_error import PoolError, PoolErrorCodes
+
+try:
+    from epplib.commands import Hello
+    from epplib.exceptions import TransportError
+except ImportError:
+    pass
+
+from gevent.lock import BoundedSemaphore
+from collections import deque
+
+logger = logging.getLogger(__name__)
+
+
+class EPPConnectionPool(ConnectionPool):
+    """A connection pool for EPPLib.
+
+    Args:
+        client (Client): The client
+        login (commands.Login): Login creds
+        options (dict): Options for the ConnectionPool
+        base class
+    """
+
+    def __init__(self, client, login, options: dict):
```

Comment:
Should this change from a dict to just **options? Its a dict for intentional separation (so you _have_ to pass in an 'options' variable when using this), but if this is more confusing than not I can just remove it